// @aws-sdk/client-cloudformation/dist-es/commands/DeleteStackSetCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteStackSetInputFilterSensitiveLog, DeleteStackSetOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDeleteStackSetCommand, serializeAws_queryDeleteStackSetCommand, } from "../protocols/Aws_query";
export class DeleteStackSetCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "DeleteStackSetCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DeleteStackSetInputFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteStackSetOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryDeleteStackSetCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryDeleteStackSetCommand(output, context);
    }
}
