// @aws-sdk/client-cloudformation/dist-es/commands/DeregisterTypeCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeregisterTypeInputFilterSensitiveLog, DeregisterTypeOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDeregisterTypeCommand, serializeAws_queryDeregisterTypeCommand, } from "../protocols/Aws_query";
export class DeregisterTypeCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "DeregisterTypeCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DeregisterTypeInputFilterSensitiveLog,
            outputFilterSensitiveLog: DeregisterTypeOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryDeregisterTypeCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryDeregisterTypeCommand(output, context);
    }
}
