// @aws-sdk/client-cloudformation/dist-es/commands/DetectStackDriftCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DetectStackDriftInputFilterSensitiveLog, DetectStackDriftOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDetectStackDriftCommand, serializeAws_queryDetectStackDriftCommand, } from "../protocols/Aws_query";
export class DetectStackDriftCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "DetectStackDriftCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DetectStackDriftInputFilterSensitiveLog,
            outputFilterSensitiveLog: DetectStackDriftOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryDetectStackDriftCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryDetectStackDriftCommand(output, context);
    }
}
