// @aws-sdk/client-cloudformation/dist-es/commands/GetTemplateSummaryCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetTemplateSummaryInputFilterSensitiveLog, GetTemplateSummaryOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryGetTemplateSummaryCommand, serializeAws_queryGetTemplateSummaryCommand, } from "../protocols/Aws_query";
export class GetTemplateSummaryCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "GetTemplateSummaryCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: GetTemplateSummaryInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetTemplateSummaryOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryGetTemplateSummaryCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryGetTemplateSummaryCommand(output, context);
    }
}
