// @aws-sdk/client-cloudformation/dist-es/commands/ListTypeVersionsCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTypeVersionsInputFilterSensitiveLog, ListTypeVersionsOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListTypeVersionsCommand, serializeAws_queryListTypeVersionsCommand, } from "../protocols/Aws_query";
export class ListTypeVersionsCommand extends $Command {
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ListTypeVersionsCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "ListTypeVersionsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListTypeVersionsInputFilterSensitiveLog,
            outputFilterSensitiveLog: ListTypeVersionsOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListTypeVersionsCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListTypeVersionsCommand(output, context);
    }
}
