// @aws-sdk/client-cloudformation/dist-es/commands/ListChangeSetsCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListChangeSetsInputFilterSensitiveLog, ListChangeSetsOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListChangeSetsCommand, serializeAws_queryListChangeSetsCommand, } from "../protocols/Aws_query";
export class ListChangeSetsCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "ListChangeSetsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListChangeSetsInputFilterSensitiveLog,
            outputFilterSensitiveLog: ListChangeSetsOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListChangeSetsCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListChangeSetsCommand(output, context);
    }
}
