// @aws-sdk/client-cloudformation/dist-es/commands/RecordHandlerProgressCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RecordHandlerProgressInputFilterSensitiveLog, RecordHandlerProgressOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryRecordHandlerProgressCommand, serializeAws_queryRecordHandlerProgressCommand, } from "../protocols/Aws_query";
export class RecordHandlerProgressCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "RecordHandlerProgressCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: RecordHandlerProgressInputFilterSensitiveLog,
            outputFilterSensitiveLog: RecordHandlerProgressOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryRecordHandlerProgressCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryRecordHandlerProgressCommand(output, context);
    }
}
