// @aws-sdk/client-cloudformation/dist-es/commands/RegisterPublisherCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RegisterPublisherInputFilterSensitiveLog, RegisterPublisherOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryRegisterPublisherCommand, serializeAws_queryRegisterPublisherCommand, } from "../protocols/Aws_query";
export class RegisterPublisherCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "RegisterPublisherCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: RegisterPublisherInputFilterSensitiveLog,
            outputFilterSensitiveLog: RegisterPublisherOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryRegisterPublisherCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryRegisterPublisherCommand(output, context);
    }
}
