// @aws-sdk/client-cloudformation/dist-es/commands/StopStackSetOperationCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopStackSetOperationInputFilterSensitiveLog, StopStackSetOperationOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryStopStackSetOperationCommand, serializeAws_queryStopStackSetOperationCommand, } from "../protocols/Aws_query";
export class StopStackSetOperationCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "StopStackSetOperationCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: StopStackSetOperationInputFilterSensitiveLog,
            outputFilterSensitiveLog: StopStackSetOperationOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryStopStackSetOperationCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryStopStackSetOperationCommand(output, context);
    }
}
