// @aws-sdk/client-cloudformation/dist-es/commands/TestTypeCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { TestTypeInputFilterSensitiveLog, TestTypeOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryTestTypeCommand, serializeAws_queryTestTypeCommand } from "../protocols/Aws_query";
export class TestTypeCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFormationClient";
        const commandName = "TestTypeCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: TestTypeInputFilterSensitiveLog,
            outputFilterSensitiveLog: TestTypeOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryTestTypeCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryTestTypeCommand(output, context);
    }
}
